### ISSUE TITLE
[BUG] Rotation/lift off is too quick

### ISSUE THREAD

**Issue by sef3613** [flybywiresim/a32nx#2817]
**Mod Version**

Latest custom FBW dev verison.

**Describe the bug**

While the flight model is much better with the custom FBW, the plane "jumps" off the ground very quickly right as rotation starts. Additionally, the rotation of the nose seems too quick with too little sidestick input required. 

**To Reproduce**

1. Do a normal takeoff in the custom FBW version.
2.
3.

**Expected behavior**

Nose slowly raises and eventually, airplane lifts off.

**Actual behavior**

Nose quickly raises and airplane lifts off almost immediately. 

**References**

Comparison with FSLabs. Same conditions, weight, and CG. Note that the FSLabs requires sidestick input up to about 10 degrees to get the nose off the ground whereas the FBW starts rotating with about 5 degrees of sidestick deflection. Also note that the FBW is at 60 feet RA just when the FSLabs achieves a positive rate of climb.

https://www.youtube.com/watch?v=BWKbU_9yIuE&feature=youtu.be

**Additional context**

Was this working before/when did the issue start occurring?
Always an issue.
Is this a problem in the vanilla unmodded game?
Yes

Discord username (if different from GitHub):

**Comment by hotshotp** [comment]
hello - this has been a reported issue with legacy flight model. when changing to modern flight model the issue is mitigated. are you on modern or legacy?

**Comment by sef3613** [comment]
@hotshotp I'm pretty sure I'm on the modern flight model as I haven't ever messed with flight model settings in MSFS. I'll check though

**Comment by sef3613** [comment]
@hotshotp Modern flight model was on in that video. It's better than the rotation without custom fbw but still too quick and "light" as you see

**Comment by MisterChocker** [comment]
duplicate of #2587

**Comment by hotshotp** [comment]
> duplicate of #2587

But that issue was wrong though - we confirmed with real pilots

**Comment by MapleNAV-TCASFAULT** [comment]
The elevator authority right now without dampening is too aggressive I find, even with the custom FBW, not being cautious can easily stall or severely oscillate the aircraft on takeoff (and yes I’m using the modern flight model), so I do agree with @sef3613, the rotation characteristics of the a32nx is a bit scuffed.

**Comment by sef3613** [comment]
Its not just the rate of rotation, but also that the main wheels lift off the ground almost instantly, like a C172 takeoff. If you watch the FSlabs or any real airbus video, you can see that the main wheels don't come off the ground until a fairly significant nose up pitch.

**Comment by donstim** [comment]
One needs to be very careful comparing two different airplane/engine combinations being flown on 2 different simulator platforms.   There are differences in controller implementations (sensitivities, etc.) as well as how ground effect is modeled.  I understand and appreciate the comments, but we choose to work off of substantiated reference data.

I believe there are two aspects at play here: (1) the airplane pitch rate, and (2) the  rate-of-climb away from the runway.  Although the initial pitch rate can be very dependent on the pitch input and the controller responsiveness in each sime, one can use the the video to examine the pitch rate. Both airplanes began their rotation at about 43 seconds in (maybe a fraction of a second  later for the FSLlabs one). The A32NX reaches 10 degrees at 47 seconds, while the FSLabs gets there at 48 seconds. 10 degrees in 4 seconds is a 2.5 deg/sec pitch rate, while 10 degrees at 5 seconds is a 2 deg/sec pitch rate. Airbus recommended procedures call for targeting a 3 deg/sec initial pitch rate.

![Screenshot (538)](https://user-images.githubusercontent.com/70166617/103436719-c2019b80-4bd3-11eb-805d-1d2ff63ba849.png)


The first second after liftoff, the pitch rate for the A32NX looks like it is a bit high, while the FSLabs looks like it might be a bit on the low side. But, at least as flown in the video, the A32NX pitch rate dampens quickly within the next second or two, although again, all of this is very subject to pilot technique and controller characteristics within the respective simulator.

For the very initial part of the climb away from the runway, we are somewhat at the mercy of the ground effect hard coded into MSFS. W don't know yet exactly what is in there and how best to accommodate it. What I will say is that the takeoff speeds, stall lift margins at those speeds, and climb rates are well matched to the real airplane performance data.

In summary, in order to make any changes to the current mod in this area, I think we would need to have substantiated reference data with specific performance targets.

**Comment by sef3613** [comment]
@donstim I understand, and I agree that the MSFS ground effect is likely what is causing the "C172-style" lift off on the mod. I know the FSlabs is in a different sim, but I've heard countless pilots praise the FSLabs flight dynamics and they say it rotates, flies, and lands very close to the real airbus. I just find that the FBW feels like it weighs 5,000 lbs when in ground effect and then feels like a proper weight once it is out of ground effect. If this is something only Asobo has control over, I hope they listen to the FBW team and keep working on it. Love the mod, great work!

**Comment by sef3613** [comment]
Sure, the FSLabs was under-rotated but do you really think the right video (MSFS) looks like an airbus rotation? It springs off the ground instantaneously. I know the FSLabs isn't perfect according to the manual, but just logically watch the video and think "does that really look correct?" for the one on the right.

**Comment by sef3613** [comment]
And the little nose drop in MSFS was the nose dropping as I left ground effect even as I was holding back pressure.I think ground effect in MSFS is a little over-exaggerated still.

**Comment by MapleNAV-TCASFAULT** [comment]
I was watching V1’s stream a few days back and he was caught of guard by how sensitive the aircraft was on rotation. 
https://youtu.be/RYnAe1vTM3o @39:31

**Comment by sef3613** [comment]
Yes, you CAN gently rotate the FBW mod but you have to use far too little side stick input if you have your sensitivity set to feel good during normal flight in the A32nx.

**Comment by sef3613** [comment]
I know that's not data from a manual but I think if people all are finding it is too quick compared to what they're used to in other sims and rw, maybe it's worth looking into.

**Comment by lukecologne** [comment]
Keep in mind that this can also be caused by the FBW. On ground, the sidestick controls the elevators directly, but on takeoff above 75kts the maximum elevator deflection is reduced from 30° do 20°. It could be that the default MSFS FBW doesn't do this, so we should revisit this once we have our own FBW system.

**Comment by MisterChocker** [comment]
> I was watching V1’s stream a few days back and he was caught of guard by how sensitive the aircraft was on rotation.
> https://youtu.be/RYnAe1vTM3o @39:31

Don't wonna say anything offensive but looked like he used improper technique, pulling on the stick, letting stick completely go, pulling again and centering again.

Just to give an example, The FSLabs320 takes off differently than the FF320.

FF320 heels and behaves similar to our 320.
And I would generally go with X-Plane flight model over anything else.

**Comment by MisterChocker** [comment]
> Keep in mind that this can be caused by the FBW. On ground, the sidestick controls the elevators directly, but on takeoff above 75kts the maximum elevator deflection is reduced from 30° do 20°. It could be that the default MSFS FBW doesn't do this, so we should revisit this once we have our own FBW system.

agree. However we might need a little bit of tweaking of possible on our own FBW since this is also reported on our FBW.

**Comment by sef3613** [comment]
@MisterChocker yeah I've heard multiple airbus pilots say that the FSLabs has a more realistic flight model than the FFa320 and toliss.

**Comment by sef3613** [comment]
I've heard the FFA320 rotation is actually really unrealistic from 2 different pilots

**Comment by MisterChocker** [comment]
You also need to keep in mind the real side stick simulates forces. I doubt any here has such a system at home. So the feel is completely off. There is no comparison of the feel from any PC sim side stick airbus to the real deal.

**Comment by sef3613** [comment]
@MisterChocker No, but you can still set sensitivities to feel correct. I'm telling you, the FFA320 should not be a comparison for rotation as it is really off too

**Comment by MapleNAV-TCASFAULT** [comment]
Wait what, the a320 sidesticks have force-feedback? I thought only the a350 had such feature.

**Comment by sef3613** [comment]
Also apparently the Warthog feels very similar in force to the real side stick which is what V1 Sim uses

**Comment by MisterChocker** [comment]
We haven't taken any other sim plane as comparison we took real life data.

**Comment by sef3613** [comment]
Well, the real life data has a rotation that's unrealistic

**Comment by sef3613** [comment]
Don't know how else to put it

**Comment by MisterChocker** [comment]
Dude are you listening to us?? We simply cannot model the rotation, the model doesn't care about the sidestick, it uses AOA nad generates different lifts at these...

**Comment by sef3613** [comment]
Also the a320 has no force feedback lol

**Comment by MapleNAV-TCASFAULT** [comment]
Well I did make a PR before that got reverted to fix such issue, I can possibly open it and refine it more following true airbus performance charts.

**Comment by sef3613** [comment]
It feels almost identical to the Thrustmaster Warthog hence why so many irl airbus pilots use the warthog. No force feedback but okay

**Comment by MisterChocker** [comment]
This is getting way too heated here..

**Comment by MapleNAV-TCASFAULT** [comment]
AFAIK, the sensitive pitch is due to the pitch authority, as when I turn off all f-ctrl computers on the left side (which should give me direct control) the aircraft is way too responsive to pitch inputs, and it feels even like a Cessna sometimes.

**Comment by sef3613** [comment]
I mean, I think a large part of the issue is the MSFS ground effect as I was told above. I hope they work with you guys and/or other devs to fix it

**Comment by MisterChocker** [comment]
Okay I am playing this card and bringing in the expert.
I am summarizing the issue for you so you don't have to read any comment in here since they are very repetitive, but feel free to do so.
The issue seems that the response of aircraft in relation to sidestick deflection is unrealistic and wrong. More precise the pitch rotation.
I know we've had this discussion internally where we came to the conclusion that this is very dependent on the hardware and software settings for the axis and we may need a config to solve this.

@aguther can you please give as an answer to the open questions and assumptions made in this issue?

I know you have very little time for things like this, I would highly appreciate the effort.

I am most likely going to lock the conversation since the discussion is too heated, please feel free to go to unlock if it's inappropriate. Discussion like that should take place on our discord sever and not on github.

**Comment by tareksabet** [comment]
@sef3613 so what's appearing from your issue that the plane lifts off instantly on rotate speed.

But this issue has lots of factors such as the CG of the aircraft, weight of the aircraft, your V speeds, your trim and your controller settings. 

So to check this issue I will do an extensive testing on it with another pilot with all possible configurations and check it. 

But what I saw in your comments above that you compared this plane which is an A320Neo to the FSLabs A320 which an A320Ceo so lots of deferences.

To finalise, I have a question for you which is : what was the CG of your plane, its weight, your V speeds and the trim you set ?
Answering this question would help us when we will test it.

Thanks for reporting this issue.

Tarek

**Comment by sef3613** [comment]
I don't remember exactly, but the cg and weight of both planes was identical. I believe the CG was 29.1 with the trim set accordingly, and the gross weight was roughly 147500 lbs. I used the v speeds from the FBW MCDU and the FSLabs arc performance which yielded almost identical numbers. Again, I'm not trying to say the FSLabs is perfect and I understand that it's a CEO, but thank you for noticing that the FBW also doesn't seem completely correct.

**Comment by sef3613** [comment]
Also, my apologies for sounding heated. I didn't mean to get heated but I realize my wording and tone came across that way over text. Like I said, I really do love this mod and I think you guys are the single best thing happening in MSFS right now. Cheers

**Comment by donstim** [comment]
I've done some looking into what parameters that we have control over (not considering our own FBW) that affect the lift-off after rotation issue:  including varying the elevator size, horizontal tail size, elevator effectiveness, the pitch stability, lift coefficient, and controller sensitivity settings. The tests were subjective with no recording of any data, so take accordingly. Also, I only conducted a couple of takeoffs with each setting that I tried, and these were at the default weight and cg.

One of the more interesting findings was that changing the horizontal tail area and elevator area had no effect at all. Even with values of 0, there was no discernible change in takeoff flight characteristics.

I felt that changing the lift coefficient was somewhat effective, but it affects the behavior more after leaving ground effect than  while in ground effect. With the current state of what I know about the flight model, it would be difficult to make a change to improve the takeoff flight characteristics without adversely affecting other areas that are now well tuned.  Part of this is due to the incorrect treatment of slats by the MSFS flight model.  However, if warranted, it is something I could continue to pursue.

Changing the pitch scalar had little to no effect. Changing the elevator effectiveness can affect this issue, but there is so much more elevator deflection available than is actually needed, you end up just requesting more elevator. Then, you can easily overshoot your pitch target and have a difficult time getting back to  where you want to be. The real airplane actually has more elevator deflection available than what is currently represented in the flight model cfg file.

What I considered to be the most effective way to address this issue at the present time is to adjust controller sensitivity settings. I have a Thrustmaster T.Flight HOTAS X joystick for which  I had been using the default sensitivity of -50% and no dead zone. I found that setting sensitivity to -70% and dead zone to 7% worked well for this.

**Comment by MapleNAV-TCASFAULT** [comment]
I also use the extremity dead-zone and reactivity feature that was recently added, I set the extremity dead-zone to 50% and the reactivity to 10% and it was way easier to fly. Note that these settings are terrible with custom FBW (takes forever to flare past the horizon), but they work really well with the default FBW.

**Comment by MapleNAV-TCASFAULT** [comment]
Ok, sorry to bother with this issue again. I'm right now tinkering with the flight_model.cfg and I have come to an answer, where the aircraft will rotate around 3 degrees/sec with 1/3 upstick input. I reduced the scalar of elevator_maxangle_scalar by 1/2 and the elevator effectiveness to 0.2. It does work pretty well coming in for a stabilized approach, but if you approach too fast, the plane tends to spring back up into the air. Also the edit to the cfg doesn't work really well with the custom FBW branch.

**Comment by MisterChocker** [comment]
> elevator effectiveness to 0.2.

Reducing this value will have negative effect on flight performance.

**Comment by donstim** [comment]
> Ok, sorry to bother with this issue again. I'm right now tinkering with the flight_model.cfg and I have come to an answer, where the aircraft will rotate around 3 degrees/sec with 1/3 upstick input. I reduced the scalar of elevator_maxangle_scalar by 1/2 and the elevator effectiveness to 0.2. It does work pretty well coming in for a stabilized approach, but if you approach too fast, the plane tends to spring back up into the air due to the overdone ground effect (i'm not sure if the ground effect values can be changed or not). Also the edit to the cfg doesn't work really well with the custom FBW branch.

Are you saying ground effect is worse with these changed values? Maybe because it forces you to use more elevator deflection due to the reduced effectiveness? 

As far as max elevator deflection angle goes, I don't think that will have much effect because hardly any deflection is normally needed (maybe a quarter of what is available) but maybe with the reduced effectiveness?

**Comment by MisterChocker** [comment]
Meta issue for tracking #3105

**Comment by aguther** [comment]
One additional info:
I discovered that the neo actually has a rotation law. I put this on my list to be implemented for the custom FBW and it will streamline the rotation performance because it’s basically a pitch rate law.

**Comment by donstim** [comment]
With the difficulty I am having in trying to find an acceptable aerodynamic solution, the custom FBW sounds like the best bet. It would override any solution I found anyway (but I'm still looking!). I am glad the airplane has FBW and doubly glad that you are implementing an accurate version of it!

**Comment by aguther** [comment]
A first implementation of the rotation law together with improvements in the C* law are now available on the fbw branch (can be installed using the new installer and choosing the experimental version). Please try it and note any feedback here.

**Comment by aguther** [comment]
The rotation law got positive feedback so I will go forward and close this issue. Feel free to reopen it when you think the rotation law does not solve the problem.

**Comment by lionhell** [comment]
_"Feel free to reopen it when you think the rotation law does not solve the problem."_
 
Hi ! Since few days/weeks I'm getting problems with rotation and A32NX dev version.
Formerly I got no problems, all was fine, never broke airplane at rotation before.
But now I'm scared every time I takeoff ! Don't know the reason but let say 1 on10 times, at rotation plane is overstressed and breaks !
Very annoying because I don't know why and more over... it is not everytime and never happened before !
Sor for me situation is worst than previously (let say FS SU5)... any clue ?

**Comment by aguther** [comment]
No idea, we did not change anything in that.

**Comment by aguther** [comment]
Can you provide me with an FDR file when this happened?

**Comment by donstim** [comment]
Sure would like to see a video of this. I've tried numerous times at different weights and speeds and cannot pull nearly enough g at rotation to overstress the airplane.

**Comment by lionhell** [comment]
OK I’m « lucky »… on the 1st attempt the plane broke at rotation 😊

I sent you the video with « grosfichiers.com » I hope you received it.

 

Regards,

 

Lionel

 

 

  _____  


 <https://home.mcafee.com/utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=emailclient?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=emailclient> 

Scanned by McAfee <https://home.mcafee.com/utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=emailclient?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=emailclient>  and confirmed virus-free.

**Comment by donstim** [comment]
Where did you send the video?  Can you just post a link here?

**Comment by lionhell** [comment]
Well. 

 

The transfer site should have sent you the link, at this mail address, I got
the confirmation.

 

No problem abyway, here is the link :

 

www.grosfichiers.com/i7sPWmfEB3n <https://www.grosfichiers.com/i7sPWmfEB3n>


 

Lionel

 

 

 

  _____  


 
<https://home.mcafee.com/utm_medium=email&utm_source=link&utm_campaign=sig-e
mail&utm_content=emailclient?utm_medium=email&utm_source=link&utm_campaign=s
ig-email&utm_content=emailclient> 

Scanned by McAfee
<https://home.mcafee.com/utm_medium=email&utm_source=link&utm_campaign=sig-e
mail&utm_content=emailclient?utm_medium=email&utm_source=link&utm_campaign=s
ig-email&utm_content=emailclient>  and confirmed virus-free.

**Comment by aguther** [comment]
The problem is that you're not corresponding with a real mailbox, you're sending it to the GitHub ticket system which is not able to process it. I guess it was not able to map it to the ticket.

Beside that while it might be very convenient for you using e-mail to respond, its often very hard to read in the ticket. The maintainers don't use e-mail and what GitHub is putting in from the mail is as I said hard to read. I would even disable it if I could. So if possible please respond directly on GitHub.

The resolution of the video is not so good, but if I spotted it right I've seen negative values for thrust reduction and acceleration heights. That's also the reason why the FMA behaves not normal during take-off. It things you're already above the bespoke altitudes. Did you enter them or did they come up automatically?

To the other thing, well it's just strange...I don't know why the sim thinks there was a crash. Beside the FMA stuff the take-off looks quite normal.

Can you provide a link to the FDR file of that take-off? Maybe we see something in there. The FDR files reside in the work folder and are named by UTC time stamps. You find the right work folder according to this guide ("Reporting issues"): https://docs.flybywiresim.com/fbw-a32nx/feature-guides/autopilot-fbw/#reporting-issues

**Comment by lionhell** [comment]
_It things you're already above the bespoke altitudes. Did you enter them or did they come up automatically?_
 
The values are automatically set, I did not enter it. If you want I can sent a video with much better quality... I chose this one because at beginning I thought I would sent to youtube and it is a veeeery long time to upload ! But since I can sent via transfer site I can rework video with better resolution if you want.
 
Regarding the plane crash, I dont understand. Fews weeks ago it NEVER happened ! I got this after some update, don't know which one ! FS ? FBW ? I dont know. But it never happened before FS  sim update 5.
 
Question: I'm not used at all with FBW procedures !!! What is FDR file ?

**Comment by aguther** [comment]
Ok, what airport did you take-off from? I only spotted 30R on the video. Then I can try if I also get the wrong values for the thrust reduction and acceleration heights.

Well, I'm not aware of anything that changed in the FBW or flight model in the questionable time frame. Last change in flight model is 2 months ago and of FBW behavior is 20 days which was about AP tuning. So nothing I can see from that side that could have triggered it.

Here you see how the work folder looks like and the FDR files:
![image](https://user-images.githubusercontent.com/6629914/130778472-af52db67-f2ae-491c-b31e-4c521b572247.png)

The FDR (Flight Data Recorder) files are recording around 400 parameters from environment and FBW, AP, ATHR, Engines so when we get them we can do some analysis on it.

**Comment by lionhell** [comment]
I’ll answer at the end of afternoon.
I departed from Montpellier LFMT.


NB: envoyé depuis mobile

**Comment by lionhell** [comment]
OK I'm back home.
 
As I said I departed from LFMT (Montpellier).
Attached below the FDR file, I hope it is the good one.

Lionel
[Uploading 2021-08-25-00-10-40.zip…]()

**Comment by lionhell** [comment]
Hum... problem with attached file, 14Mo may be too big ?
 
Then the FDR file can be downloaded here: https://www.grosfichiers.com/cUnTda8vB45

Lionel

**Comment by obsidian-mint** [comment]
I do have the exact same Issue as described here. Funfact: If I spawn the Plane at any runway (Warm and Bright, not Cold and Dark) and just TOGA then i DON'T get any problems, completely normal behaviour so i assume that something with the loading / cg etc. might be the problem. Thats not all. When i land at any runway directly at touchdown I get an CTD. What do you think? Should I make an new Issue?
